File: 백준 10845번 큐.cpp
#include<iostream>
#include<algorithm>
#include<queue>
using namespace std;

int main()
{
	int n;
	cin>>n;
	queue <int>q;
	int count(0);
	for(int i=0; i<n; i++)
	{
		string s;
		cin>>s;
		if(s=="push")
		{
			int av;
			cin>>av;
			q.push(av);	
		}
		else if(s=="pop")
		{
			if(q.size()==0)
				count =-1;
			else
			{
				count=q.front();
				q.pop();
			}
			cout<<count<<"\n";
		}
		else if(s=="empty")
		{
			if(q.size()==0)
			count=1;
			else
			count=0;
			cout<<count<<"\n";
			
		}
		else if(s=="size")
		cout<<q.size()<<"\n";
		else if(s=="front")
		{
			if(q.size()==0)
			count=-1;
			else
			count=q.front();
			cout<<count<<"\n";
		}
		else if(s=="back")
		{
			if(q.size()==0)
			count=-1;
			else
			count=q.back();
			cout<<count<<"\n";
		}
			
	}
}

File: Boj-Kotlin&Java/백준 10845번 큐.kt
package boj14681kot

import java.util.*
import java.util.Queue as Queue

fun main(args:Array<String>)
{
    val br=System.`in`.bufferedReader()
    val bw=System.out.bufferedWriter()
    val n=br.readLine().toInt()
    val queue:Queue<Int> = LinkedList<Int>()
    var tmp:Int?=0
    for(i in 0 until n step 1)
    {
        val string=br.readLine().split(" ")
        if(string[0].equals("push"))
        {
            queue.offer(string[1].toInt())
            tmp=string[1].toInt()
        }else if(string[0].equals("front"))
        {
            if(queue.isEmpty())
                bw.write("-1\n")
            else
                bw.write("${queue.peek()}\n")
        }
        else if(string[0].equals("size")){
            bw.write("${queue.size}\n")
        }else if(string[0].equals("empty"))
        {
            if(queue.isEmpty())
                bw.write("1\n")
            else
                bw.write("0\n")
        }else if(string[0].equals("back"))
        {
            if(queue.isEmpty())
                bw.write("-1\n")
            else
                bw.write("${tmp}\n")
        }else if(string[0].equals("pop"))
        {
            if(queue.isEmpty())
                bw.write("-1\n")
            else
                bw.write("${queue.poll()}\n")
        }
    }
    bw.flush()
}